web/github.go

```diff
@@ -70,45 +131,153 @@ func (g *GitHub) Orgs(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return nil, fmt.Errorf(http.StatusText(http.StatusInternalServerError))
+	}
+
+	// a GitHub App can also be installed in a User account
+	if installation.GetAccount().GetType() != "Organization" {
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+	}
+
+	admin, err := g.admin(w, r, installation, login)
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed to check user admin role")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return nil, fmt.Errorf(http.StatusText(http.StatusInternalServerError))
+	}
+
+	if !admin {
+		// 404 to avoid leaking to a non admin if the org exists or is installed
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+	}
+
+	return installation, nil
+}
+
+// orgResponse is the response type used by the individual organization handler
+type orgResponse struct {
+	ID     int64  `json:"id"`
+	Name   string `json:"name"`
+	Config string `json:"config"`
+}
+
+// Org writes in the response the individual organization requested by the URL
+// parameter "orgName", only if the user is an admin
+func (g *GitHub) Org(w http.ResponseWriter, r *http.Request) {
+	installation, err := g.orgInstallation(w, r)
+	if err != nil {
+		return
+	}
+
+	idStr := strconv.FormatInt(installation.GetAccount().GetID(), 10)
+	config, err := g.OrganizationOp.Config(r.Context(), github_provider.Provider, idStr)
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed read the organization config from the DB")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	successJSON(w, r, orgResponse{
+		ID:     installation.GetAccount().GetID(),
+		Name:   installation.GetAccount().GetLogin(),
+		Config: config,
+	})
+}
+
+type updateOrgReq struct {
+	Config string `json:"config,omitempty"`
+}
+
+// UpdateOrg is a hander that updates the organization settings, and returns
+// the updated organization information with the same response as Org
+func (g *GitHub) UpdateOrg(w http.ResponseWriter, r *http.Request) {
+	var configRequest updateOrgReq
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed to read the request body")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	err = json.Unmarshal(body, &configRequest)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Bad Request. Body is not a valid JSON: %s", err), http.StatusBadRequest)
+		return
+	}
+
+	installation, err := g.orgInstallation(w, r)
+	if err != nil {
+		return
+	}
+
+	// TODO: check config is valid yaml
```

Comment:
let's fix this TODO before merge. UI is just a text field. It's very easy to send incorrect input.

Reply:
Done in 928e9b1.